src/file/exceptions.php

```diff
@@ -12,23 +12,20 @@
 
 use namespace HH\Lib\Experimental\{IO, OS};
 
-abstract class Exception
+/**
+ * An exception thrown when opening a file fails.
+ */
+final class OpenException
```

Comment:
`OpenException` seems weird and the name isn't really as helpful as `LockAcquisitionException` or `HostResolutionException`

Reply:
Well the name likely to be used in code is `File\OpenException` which looks good to me.

Reply:
why did it fail to open ? non-existent file ? `NonExistingFileException`. unreadable file ? `UnreadableFileException` ... 


```hack
try {
  file = open();
} catch(MissingFileException $e) {
  create();
  file = open();
} catch(UnreadableFileException $e) {
  throw new DomainException('blah blah', $e);
}
```

see https://github.com/nuxed/filesystem/tree/master/src/Nuxed/Filesystem/Exception :wink:

Reply:
Not relevant to this PR: this PR does not add OpenException, or change how specific it is. This PR only changes the ancestors.